ando/AnDOChecker.py

```diff
@@ -0,0 +1,209 @@
+                                          format(current_mandatoryfolder_rule))
+
+    # if there are no errors, the data set is valid!
+    valid = len(error_list) == 0
+
+    return valid, error_list
+
+
+def get_list_of_rules_with(file_name_regexp, file_ext_regexp=None):
+    """
+    Function that create rules base on the regexp in rulesStructured.py
+
+    Parameters
+    ----------
+
+        file_name_regexp: list
+            list of filenames
+        file_ext_regexp: list
+            list of extensions
+
+    Returns
+    -------
+        list_of_rules : list
+            Concatenate every extension with is respective file
+
+    Examples
+    -------
+    if :
+    file_name_regexp = [
+        ['sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\\w\\-]*)_ephys'],
+        ['sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\\w\\-]*)_channels'],
+        ]
+
+    and :
+    file_ext_regexp = ['.tsv', '.json']
+
+    then :
+    list_of_rules = [
+        'sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\\w\\-]*)_ephys.tsv',
+        'sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\\w\\-]*)_ephys.json',
+        'sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\\w\\-]*)_channels.tsv',
+        'sub-([a-zA-Z0-9]+)_ses-([a-zA-Z0-9]+)([\\w\\-]*)_channels.json',
+        ]
+    """
+    list_of_rules = list()
+    if file_ext_regexp is None:
+        list_of_rules = file_name_regexp
+    else:
+        if len(file_name_regexp) == 2:
+            for filename in file_name_regexp[:-1]:
+                for extension in file_ext_regexp:
+                    list_of_rules.append(filename + str(extension))
+        else:
+            for filename in file_name_regexp:
+                for extension in file_ext_regexp:
+                    list_of_rules.append(filename + str(extension))
+
+    return list_of_rules
+
+
+def main():
+    """
+    Main file of the validator. uses other class methods for checking
+    different aspects of the directory path.
+
+    usage: checker.py [-h] [-v] path
+
+            positional arguments:
+            directory           Name of the directory that contains the data set to be checked
+
+            optional arguments:
+            -h, --help     show this help message and exit
+            -v, --verbose  increase output verbosity
+
+
+    """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--verbose', action='store_true',
+                        help='increase output verbosity')
+    parser.add_argument('directory', help='Name of the directory to be checked')
+    args = parser.parse_args()
+
+    try:
+        directory = args.directory
+    except IndexError:
+        directory = '.'
+
+    if not directory:
+        print('Directory does not exist:', directory)
+        exit(1)
+    dataset_validity, error_list = is_valid(directory)
+    if dataset_validity:
+        print("Congratulations!\n" +
+              directory +
+              " respects the ephys-BIDS specifications")
```

Comment:
```suggestion
        print("Congratulations!\n"
            f"{directory} respects the ephys-BIDS specifications")
```

Reply:
since this is a new package we can use the new Python string formatting features ;) (and also in the other prints below...)

Reply:
Done